odl/tomo/operators/ray_trafo.py

```diff
@@ -321,8 +344,13 @@ def _call(self, x, out=None):
                 return astra_cpu_back_projector(x, self.geometry,
                                                 self.range, out)
             elif data_impl == 'cuda':
-                return astra_cuda_back_projector(x, self.geometry,
-                                                 self.range, out)
+                backproj = getattr(self, 'astra_backprojector', None)
+                if backproj is None:
+                    self.astra_backprojector = AstraCudaBackProjectorImpl(
+                        self.geometry, self.range, self.domain,
+                        use_cache=self.use_cache)
+
+                return self.astra_backprojector.call_backward(x, out)
```

Comment:
Well I was guessing that it was intended for making it one class but rather that had been the goal once and then abandoned. It's totally fine in the current case.